File: android/app/src/main/java/com/dailymemory/domain/usecase/reminder/CompleteReminderUseCase.kt
package com.dailymemory.domain.usecase.reminder

import com.dailymemory.domain.repository.ReminderRepository
import java.time.LocalDateTime
import javax.inject.Inject

/**
 * Use case for marking a reminder as complete (triggered).
 */
class CompleteReminderUseCase @Inject constructor(
    private val reminderRepository: ReminderRepository
) {
    suspend operator fun invoke(reminderId: String): Result<Unit> {
        return try {
            reminderRepository.markAsTriggered(reminderId, LocalDateTime.now())
            Result.success(Unit)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: ios/DailyMemory/Domain/UseCase/Reminder/CompleteReminderUseCase.swift
import Foundation

/// Use case for marking a reminder as complete (triggered).
final class CompleteReminderUseCase {
    private let reminderRepository: ReminderRepository

    init(reminderRepository: ReminderRepository = ReminderRepositoryImpl()) {
        self.reminderRepository = reminderRepository
    }

    func execute(reminderId: String) async throws {
        try await reminderRepository.markAsTriggered(id: reminderId)
    }
}
